appoptics/appoptics-apm-node
# to be released eventually

v7.0.0

major changes & bug fixes
- move to `ace-context` with `cls-hooked` as backup. remove `continuation-local-storage`. [AO-14851]
- add `forceNewTrace` option to `startOrContinueTrace()` [AO-14890]
- force new context on inbound HTTP requests
- support restify v7+ [AO-14589]
- fix http RemoteURL when search/query [AO-15037]
- exit http/s client spans correctly on socket error events
- exit http/s client spans on upgrade events [AO-13795]
- update zlib to handle brotli compression [AO-11896]
- don't add Database KV when database is undefined in connect [AO-15104]

minor changes
- suppress logging of missing spans during node bootstrap phase
- add static debugging config for ace-context
- more consistent span logging messages
- optimize built-in patching [AO-14756]
- check new http_parser values during init [AO-14657]

testing changes
- test `q` and `bluebird` promises in `custom.test.js` [AO-8673]
- fix sendReport() being called twice on outer:exit event in zlib testing [AO-11898]
- add .mocharc with `exit: true` to prevent tests from hanging.
- make http/https testing use exact same code `test/probes/http-common.js`
- http/https tests now have real socket errors.
- `test/composite/request.test.js` now generates a real socket error
- add `forceNewTrace` testing to `test/custom.test.js`
- add websocket tests
- clear context after pg6-plus test
- don't test zlib's brotli before node v11.7.0

bookkeeping
- ignore *.log files in package root
- .eslintrc.yaml - add BigInt, ignore template literal max line length
- updated guides/api.md
- change form of license to "SEE LICENSE IN <link>" [AO-14757]

reviewing notes
- all these changes have been reviewed before via PRs to vNext.

[AO-8673]: https://swicloud.atlassian.net/browse/AO-8673

[AO-14851]: https://swicloud.atlassian.net/browse/AO-14851
[AO-14890]: https://swicloud.atlassian.net/browse/AO-14890
[AO-14589]: https://swicloud.atlassian.net/browse/AO-14589
[AO-15037]: https://swicloud.atlassian.net/browse/AO-15037
[AO-13795]: https://swicloud.atlassian.net/browse/AO-13795
[AO-11896]: https://swicloud.atlassian.net/browse/AO-11896
[AO-14756]: https://swicloud.atlassian.net/browse/AO-14756
[AO-14657]: https://swicloud.atlassian.net/browse/AO-14657
[AO-11898]: https://swicloud.atlassian.net/browse/AO-11898
[AO-14757]: https://swicloud.atlassian.net/browse/AO-14757

[AO-15104]: https://swicloud.atlassian.net/browse/AO-15104
